src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php

```diff
@@ -49,7 +49,19 @@ protected function createController($controller)
             $controller = $this->parser->parse($controller);
         }
 
-        return parent::createController($controller);
+        $resolvedController = parent::createController($controller);
+
+        if (1 === substr_count($controller, ':') && is_array($resolvedController)) {
```

Comment:
shouldn't this be done in the parent class instead, which is the one supporting container injection ?